src/preferences/dialog/dlgprefsound.cpp

```diff
@@ -690,6 +692,14 @@ void DlgPrefSound::slotResetToDefaults() {
     SoundManagerConfig newConfig(m_pSoundManager.get());
     newConfig.loadDefaults(m_pSoundManager.get(), SoundManagerConfig::ALL);
     loadSettings(newConfig);
+
+    int keylock_engine = static_cast<int>(EngineBuffer::RUBBERBAND_FASTER);
+    int index = keylockComboBox->findData(keylock_engine);
+    DEBUG_ASSERT(index >= 0);
+    if (index >= 0) {
+        keylockComboBox->setCurrentIndex(index);
+    }
+
```

Comment:
RUBBERBAND_FASTER is always available and therefore in the combobox.
Why are these safety measures necessary?

The line below can be removed now?

Reply:
Yes the line below can be removed.

In the current version index and engine have the same integer number.
This may change when a user has engine 5 of a yet unknown future version or such. It will become  the last combo box position. Which is 2 or 3. 

That is why I have made the selection index independent from the selected engine, which is now stored as user data.
Even though I know the index already Ihere, I look it up for correctness. A kind of future Prove measure. 

I'd this OK?

Reply:
Okay that makes sense.
It' just I don't see how RUBBERBAND_FASTER could not be available. For additional scaler engines one would need to modify the enginebuffer enum in order to make them available, and if RUBBERBAND_FASTER was removed from the enum class (or renamed) `static_cast<int>(EngineBuffer::RUBBERBAND_FASTER)` wouldn't compile anyway, right?
Or is this supposed to cover the case where RUBBERBAND_FASTER is in the enum but deactivated for some reason?

Reply:
Oh sorry I think I confused you due to my wrong commit message. RUBBERBAND_FASTER will be always available (for now) 

In a future version, let's say we remove Soundtouch, than we should keep the enum value, but remove the selection from the combo box. I will add a comment that the enum values must not be changed only appended.